Ansible 4

SkipUFO opened the pull request:
# Выполнено ДЗ № 11

 - [x] Основное ДЗ
 - [x] Задание со *

## 1.1 Что было сделано
- Установлен Vagrant (https://www.vagrantup.com/downloads.html)
- Переделаны и параметризованы таски и модифицированы плейбуки для запуска инфраструктуры при помощи Vagrant на провайдере VirtualBox
- Изучено как получить под каким пользователем vagrant запускает плейбуки (vagrant ssh-config, либо заходим на хост vagrant ssh <hostname>)
- Установлена molecule (были проблемы с последней версией, поставил в requirements.txt - molecule==2.10)
- Установлен дополнительно python-pytest (иначе molecule verify падала sh.CommandNotFound: py.test)
- Переделана роль для сборки app, db образов
## 1.2 В задании со *
- Настройка nginx: если попробовать вставить код следующим образом
```
"app:vars" => { 
          "db_host" => "10.10.10.10", 
          "nginx_sites" => "default:
                            - listen 80
                            - server_name \"reddit\"
                            - location / {
                              proxy_pass http://127.0.0.1:9292;
                            }"
        }
```
то получим ошибку 
```
fatal: [appserver]: FAILED! => {"msg": "with_dict expects a dict"}
``` 
пробовал подсмотреть в репозитории jdauphant.nginx, но там используется в коде yml-теста, в итоге пришел к такому
```
ansible.extra_vars = {
      "deploy_user" => "vagrant", 
      "nginx_sites" => {
        "default": [ "listen 80", "server_name \"reddit\"", "location / { proxy_pass http://127.0.0.1:9292; }" ]
      }
    }
```
- Второе задание со * не сделал (оно также как в предыдущем задании работа с travis)

## PR checklist
 - [x] Выставил label с номером домашнего задания
 - [x] Выставил label с темой домашнего задания

SkipUFO committed:
Reorganized ansible repo:
1. Made roles for app and db
2. playbooks moved to dir playbooks
3. created environments dirs

SkipUFO committed:
Added nginx (80) port to firewall

SkipUFO committed:
Added vault usage

SkipUFO committed:
Merge branch 'ansible-2' into ansible-3

SkipUFO committed:
Moved packer yaml files

SkipUFO committed:
Removed trailing spaces

SkipUFO committed:
moved packer yaml files

SkipUFO committed:
Fixed path to provision files

SkipUFO committed:
Removed trailing white spaces

SkipUFO committed:
Removed trailing spaces

SkipUFO committed:
Added dynamic inventory to ansible environment dirs

SkipUFO committed:
Added ansible-3 README section

SkipUFO committed:
Added ignore files

SkipUFO committed:
Added definitions for vagrant infrastructure for VirtualBox

SkipUFO committed:
Added nginx config to variables

SkipUFO committed:
Added tests for db role

SkipUFO committed:
Added HW-11 description

SkipUFO committed:
Fixed \Z\n errors

SkipUFO committed:
Fixed role using mixed normal tasks

SkipUFO committed:
Added README.md description

grim-yawn commented on ansible/playbooks/packer_db.yml:
```diff
@@ -21,8 +21,4 @@
-    systemd:
-      name: mongod
-      enabled: yes
+- include: db.yml
```
Не собирается, так как в db.json не добавлен путь к ролям.

grim-yawn replied:
Дублирование, т.к. в установка mongo описана в роли db.
Не будет работать так как в db.yml указаны `hosts: db`, а в packer будет только all.

Надо убрать include и заменить его на использование роли db.

SkipUFO committed:
packer_db.yml
 - Switched to use role db, instead of tasks
 - Hosts changed to only db
packer_app.yml
 - Hosts changed to only app
